odoo/addons/base/models/ir_http.py

```diff
@@ -318,7 +318,8 @@ def _get_record_and_check(self, xmlid=None, model=None, id=None, field='datas',
             # We have prefetched some fields of record, among which the field
             # 'write_date' used by '__last_update' below. In order to check
             # access on record, we have to invalidate its cache first.
-            record._cache.clear()
+            if not record.env.su:
+                record._cache.clear()
```

Comment:
This doesn't break https://github.com/odoo/odoo/commit/0d5fcdc1c7d13f3ac1d5868d7116b5c4e45a692c so might be a correct solution

Reply:
The `if` statement can even contain the statement below.

Reply:
Suggestion: you can actually explicitly call the check methods
```python
if not record.env.su:
    record.check_access_rights('read')
    record.check_access_rule('read')
```